### DISCUSSION TITLE
可算琢磨明白开着VPN怎么连接模拟器了

### DISCUSSION THREAD

**Discussion by zeyugan** [MaaAssistantArknights/MaaAssistantArknights#1117] [General]
人在国外，开模拟器不挂VPN一分钟掉一次，挂了VPN之后app不能自动连上模拟器一直报错。今天琢磨了一下总算弄明白怎么挂着VPN长草了……

0. 启动VPN和模拟器，我用的nord（香港节点）和蓝叠
1. 手动下载一个adb sdk （https://developer.android.com/studio/releases/platform-tools）
2. 解压之后将platform-tool加入Environment Variables - Path
3. cmd（建议管理员，因为可能需要netstat）
     1. adb start-server
     2. netstat -a - b找到你的模拟器（蓝叠是5555） 
     3. adb connect localhost:5555
4. 在app中设置 - 连接设置，adb路径改为下载的platform-tool/adb.exe
5. 开始任务，可以正确连接了

试了重启一次，不需要再次手动连接adb，顺序打开VPN 模拟器和app可以直接跑任务。
~~希望没有火星~~

**Comment by MistEO** [comment]
其实可以把 vpn 开在模拟器里面，安卓 app 这种
